runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -127,7 +185,14 @@ class ExecutionContext final {
 
     [[nodiscard]] auto getDeviceRecorderStatus() const -> bool { return _tape_recording; }
 
-    [[nodiscard]] bool initDevice(std::string_view name) noexcept
+    [[nodiscard]] QuantumDevice *loadDevice(std::string filename)
+    {
+        _driver_so_ptr = std::make_unique<SharedLibraryManager>(filename);
+        void *f_ptr = _driver_so_ptr->getSymbol("getCustomDevice");
+        return f_ptr ? reinterpret_cast<decltype(getCustomDevice) *>(f_ptr)() : NULL;
```

Comment:
```suggestion
        return f_ptr ? reinterpret_cast<decltype(getCustomDevice) *>(f_ptr)() : nullptr;
```

Reply:
Changed without adding suggestion to batch.